### ISSUE TITLE
[FEATURE] enhance terraform network to not pruge route_cidr and route_gateway

### ISSUE THREAD

**Issue by lanfon72** [harvester/harvester#2261]
### The problem
When using terraform Harvester to apply **network resource** with `route_mode="auto"`, Harvester will update `route_cidr` and `route_gateway` then update _Route Connectivity_ to be **Active**.

And If user apply the resource again, terraform will consider to purge `route_cidr` and `route_gateway` as the following snapshot:
![image](https://user-images.githubusercontent.com/5169694/167812868-c04951ae-1a22-4005-b9c1-4f4962842119.png)
![image](https://user-images.githubusercontent.com/5169694/167813320-7200989f-3188-44c3-b7fd-36cbef16b096.png)
![image](https://user-images.githubusercontent.com/5169694/167813333-cb1623c0-2f09-4831-a64e-f25c458d7085.png)

### The solution
we should not purge those data generated by Harvester.

### The Alternatives

**Comment by yasker** [comment]
@futuretea I am not sure if this is expected?

**Comment by futuretea** [comment]
@yasker not expected, will cause user confusion.

**Comment by lanfon72** [comment]
@guangbochen, @rebeccazzzz, I think this will fix both on `v1.0.3` and `v1.1.0`, should we create another issue to mark it is backported?

**Comment by lanfon72** [comment]
Verified this feature has been implemented.

![image](https://user-images.githubusercontent.com/5169694/175333418-9d092b58-0202-4799-ba5b-476e02e7a785.png)


Test Information
----
* Environment: **qemu/KVM 2 nodes**
* Harvester Version: **v1.0-ec0bb4be-head** and **master-949bbb0c-head**
* **ui-source** Option: **Auto**
* **ui-index**: https://releases.rancher.com/harvester-ui/dashboard/release-harvester-v1.0/index.html (for **v1.0** only)

### Verify Steps:
1. Install Harvester with any nodes
2. Install **terraform-harvester-provider** (using [master-head](https://github.com/harvester/terraform-provider-harvester/wiki/Use-the-master-head-docker-image-for-testing) for testing)
3. Execute `terraform init`
4. Create the file `network.tf` as following snippets, then  execute `terraform import harvester_clusternetwork.vlan vlan` to import default vlan settings
```terraform
resource "harvester_clusternetwork" "vlan" {
  name                 = "vlan"
  enable               = true
  default_physical_nic = "harvester-mgmt"
}
resource "harvester_network" "vlan1" {
  name      = "vlan1"
  namespace = "harvester-public"

  vlan_id = 1
  route_mode = "auto"
}
```
5. execute `terraform apply`
6. Login to dashboard then navigate to _Advanced/Networks_, make sure the **Route Connectivity** becomes `Active`
7. Execute `terraform apply` again and more times
8. Resources should not be changed or added or destroyed.
